Plugins/DotNET/DotNETExports.cpp

```diff
@@ -331,6 +346,24 @@ static const char* StackPopString()
     return strdup(String::ToUTF8(value.CStr()).c_str());
 }
 
+static const char* StackPopRawString()
+{
+    auto vm = Globals::VirtualMachine();
+    ASSERT(vm->m_nRecursionLevel >= 0);
+
+    CExoString value;
+    if (!vm->StackPopString(&value))
+    {
+      LOG_WARNING("Failed to pop string - recursion level %i.", vm->m_nRecursionLevel);
+      return nullptr;
+    }
+
+    static std::string retVal = value.CStr();
+    LOG_DEBUG("Popped string '%s'.", value.m_sString);
+
+    return retVal.c_str();
```

Comment:
Either this one or `StackPopString()` above is wrong. The previous one assumes dotnet takes ownership of the pointer to heap memory and will free it. This one will still return a pointer to heap memory, but it does not relinquish the ownership. So second time it's called, when `= value.CStr()` is run, it might try to use or free the old memory, causing a boom.

Alternatively, the previous assumption from `StackPopString()` is wrong and it actually leaks memory, in which case it should use a `static` variable like here instead of `strdup()`. Could you verify which one it is?

Reply:
Ah, I missed that part. Alright then, sounds good. But then it does need four more entries and can't use the reserved UTF8 ones.

Just keep in mind that you have to copy the string before anyone tries to pop a second one, because the buffer is shared.